### PR TITLE
mpt project

### LINKED ISSUE #648: MPTTrie classes should be available in class library independently from StateStore plugin
As [per @ZhangTao1596](https://github.com/neo-project/neo-modules/pull/644#issuecomment-930708357), the way to validate a state service proof locally is to use MPTTrie.VerifyProof method. But the MPTTrie class is only available in StateService.dll which also depends on RpcServer plugin and Neo.ConsoleService.

Can we move the MPTTrie code into a separate library that can be used independently of StateService plugin?

### PR DESCRIPTION
**Pull request by ZhangTao1596**
Close #648

### CODE REVIEW

**Comment by ZhangTao1596** [comment]
@devhawk Names updated.

**Review by shargon — DISMISSED** [review]
Please, wait @erikzhang 's review

File: neo-modules.sln [hunk 1]
```diff
@@ -30,12 +30,14 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Neo.Plugins.OracleService.T
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "StateService", "src\StateService\StateService.csproj", "{A0F4A66F-6F87-4B99-B8BE-A779BC002F47}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Neo.Plugins.StateService.Tests", "tests\Neo.Plugins.StateService.Tests\Neo.Plugins.StateService.Tests.csproj", "{149822EC-4E0C-425F-A032-4196B615BFEB}"
-EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "DBFTPlugin", "src\DBFTPlugin\DBFTPlugin.csproj", "{90185D3E-4813-4BC1-98FE-26FD34311403}"
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Neo.Plugins.RpcServer.Tests", "tests\Neo.Plugins.RpcServer.Tests\Neo.Plugins.RpcServer.Tests.csproj", "{0DDAF738-0FD3-40A7-A433-C514BCDDF542}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "MPTTrie", "src\MPTTrie\MPTTrie.csproj", "{D167FA6B-D2A3-4D8A-A65D-686DD06650F6}"
```
Comment (erikzhang) [inline neo-modules.sln:37]: What is this project?

Comment (ZhangTao1596) [reply neo-modules.sln:37]: MPT implementation. It's used to be in StateService.

Comment (erikzhang) [reply neo-modules.sln:37]: Why? I think we can publish StateService directly.

Comment (ZhangTao1596) [reply neo-modules.sln:37]: `StateService` need `RpcClient`, `Neo.ConsoleServiceBase`, state root consensus and store.
People usually only want mpt implementation.

**Comment by erikzhang** [comment]
Can you fix the ut?

**Comment by ZhangTao1596** [comment]
> Can you fix the ut?

Sure.

**Comment by ZhangTao1596** [comment]
@erikzhang @shargon Merge?

**Review by erikzhang — APPROVED** [review]

**Review by shargon — APPROVED** [review]

**Pull request merged**